core-blocks/archives/editor.scss

```diff
@@ -0,0 +1,4 @@
+.gutenberg .wp-block-archives a {
+	pointer-events: none;
```

Comment:
Unfortunately this doesn't completely disable the link. A user is still able to "click" it by focusing it using their keyboard and then pressing ENTER.

I'd try wrapping the `<ServerSideRender>` component with the `<Disabled>` component that we have and seeing what happens.

https://github.com/WordPress/gutenberg/tree/master/components/disabled